sklearn/base.py

```diff
@@ -578,3 +580,39 @@ def is_regressor(estimator):
         True if estimator is a regressor and False otherwise.
     """
     return getattr(estimator, "_estimator_type", None) == "regressor"
+
+
+def frozen_fit(estimator, method, X, y, **kwargs):
+    """Fit the estimator if not frozen, and return the result of method
+
+    A frozen estimator has an attribute ``frozen`` set to True
+
+    Parameters
+    ----------
+    estimator
+    method : str
+        One of {'fit', 'fit_transform', 'fit_predict'} or similar.
+    X
+    y
+        will only be passed when fitting
+    kwargs
+        will only be passed when fitting
+
+    Returns
+    -------
+    out
+        estimator if ``method == 'fit'``, else the output of ``transform`` etc.
+        If the estimator has attribute ``frozen`` set to True, it will not be
+        refit.
+    """
+    if getattr(estimator, 'frozen', False):
+        if method == 'fit':
+            return estimator
+        if not method.startswith('fit_'):
+            raise ValueError('method must be "fit" or begin with "fit_"')
+        method = getattr(estimator, method[4:])
```

Comment:
Wow this is a bit of a hack lol. And a model doesn't necessarily have that, right? If I call ``fit_transform`` on a frozen T-SNE, it's gonna give me an attribute error. Not sure if there's a better solution though.